### PR TITLE
Suggested variables when undefined, and more

### LINKED ISSUE #19: Log Levels
Now that lager is started up by the cuttlefish_escript. We'll have to be better about what gets logged. Pretty much everthing is at 'info' currently.

We should move everything to debug. Have one 'info' level message to say "This is enabled" which doesn't output if it's using old style config..

Then we should use "warnings" for any config red flags and "errors" for anything that won't generate a config file.

### PR DESCRIPTION
**Pull request by joedevivo**
A couple of little fixes, but let's start with the headlining feature.
### sugestions when variable doesn't exist

I tried setting `anti_entrophy = on` in my riak.conf, this was the output

```
➜  riak_ee git:(develop) ✗ ./rel/riak/bin/riak config generate
10:57:23.202 [warning] You've tried to set anti_entrophy, but there is no mapping
10:57:23.202 [warning]   it could be a typo. Did you mean one of these?
10:57:23.218 [warning]     anti_entropy
10:57:23.218 [warning]     anti_entropy.tick
10:57:23.218 [warning]     anti_entropy.expire
```

Oh yeah, suggestions when you make a mistake! This addresses #15 but in a larger than life kind of way

This PR also fixes #19 and #47, which were pretty much the same issue anyway.

Also, @reiddraper noticed a broken test, which I broke in previous merge, but the functionality was correct, I just had to alter the test to compensate.

### CODE REVIEW

File: src/cuttlefish_generator.erl [hunk 1]
```diff
@@ -313,9 +313,29 @@ transform_datatypes(Conf, Mappings) ->
...
-                    %% this log message is also called recursively doh!
-                    %%lager:warning("No setting defined for ~s", [Key]),
+                    %% So, this error message isn't so performant (s/o @argv0)
+                    %% but it shouldn't happen too often, and I think it's important
+                    %% to give users this feedback.
+
+                    %% It won't prevent anything from starting, but will let you know
+                    %% that you're trying to set something that has no effect
+                    VarName = string:join(Variable, "."),
+                    lager:warning("You've tried to set ~s, but there is no mapping", [VarName]),
```
Comment (seancribbs) [inline src/cuttlefish_generator.erl]: mapping -> "setting of that name". Users don't know/care about mappings.

Comment (joedevivo) [inline src/cuttlefish_generator.erl]: ha, I think I prefer "You've tried to set ~s, but that's not a real thing"

File: src/cuttlefish_generator.erl [hunk 2]
```diff
@@ -313,9 +313,29 @@ transform_datatypes(Conf, Mappings) ->
...
+                    lager:warning("  it could be a typo. Did you mean one of these?"),
+
+                    Possibilities = [ begin
+                        MapVarName = string:join(cuttlefish_mapping:variable(M), "."),
+                        {cuttlefish_util:levenshtein(VarName, MapVarName), MapVarName}
+                    end || M <- Mappings],
+                    Sorted = lists:sort(Possibilities),
+                    NumSuggest = 3,
+                    Top = case length(Sorted) > NumSuggest of
```
Comment (seancribbs) [inline src/cuttlefish_generator.erl]: Just use `lists:sublist/2` here.

``` erlang
7> lists:sublist([], 3).
[]
8> lists:sublist([a,b], 3).  
[a,b]
9> lists:sublist([a,b,c], 3).
[a,b,c]
10> lists:sublist([a,b,c,d,e], 3).
[a,b,c]
```

**Comment by seancribbs** [comment]
:+1: nice addition :smile_cat:

**Comment by reiddraper** [comment]
Very cool, I'm always impressed when software suggests corrections for spelling mistakes I make.

**Pull request merged**